File: python/704_BInary-Search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        st, ed = 0, len(nums)
        if ed == 1:
            return st if nums[st] == target else -1

        pivotIdx = ed/2
        pivot = nums[pivotIdx]

        while(pivot!=target):
            if ed - st <= 1:
                return  st if nums[st] == target else -1
            elif pivot > target:
                ed = pivotIdx
            else:
                st = pivotIdx
            
            pivotIdx = (st+ed)/2
            pivot = nums[pivotIdx]
        
        return pivotIdx

File: c/704_Binary-Search.c
int search(int* nums, int numsSize, int target){
    if(numsSize == 1) return (nums[0] == target) ? 0 : -1;

    int st = 0;
    int ed = numsSize;
    int pivotIdx = (st+ed)/2;
    int pivot = *(nums+pivotIdx);

    while(ed-st>1){
        if (pivot == target)
            return pivotIdx;
        else if (pivot>target)
            ed = pivotIdx;
        else if (pivot<target)
            st = pivotIdx;
        
        pivotIdx = (st+ed)/2;
        pivot = *(nums+pivotIdx);
    }
    return (nums[st] == target) ? st : -1;
}